### PR TITLE
feat: #Task:229802 Creation of Student Login Page in React

### PR DESCRIPTION
**Pull request by deepali-chavhan**

### CODE REVIEW

File: src/pages/auth/Login.tsx [hunk 1]
```diff
@@ -1,67 +1,278 @@
...
+          //   localStorage.setItem("fcmToken", "");
+          // }
+          // eventBus.publish("AUTH", {
+          //   eventType: "LOGIN_SUCCESS",
+          //   data: {
+          //     token: token,
+          //   },
+          // });
+          navigate("/home");
+          window.location.reload();
```
Comment (sagarkoshti1990) [inline src/pages/auth/Login.tsx:172]: Use `navigate(0)` instead of `window.location.reload()`.

File: src/pages/auth/Login.tsx [hunk 2]
```diff
@@ -1,67 +1,278 @@
...
+          </Button>
+
+          {errors.alert && (
+            <Alert status="error" mt={4}>
+              <AlertIcon />
+              <AlertTitle>{errors.alert}</AlertTitle>
+            </Alert>
+          )}
+
+          <InfoModal
```
Comment (sagarkoshti1990) [inline src/pages/auth/Login.tsx:266]: If possible, rename `InfoModal` to `PopupModal`.

File: src/services/auth/auth.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      'Content-Type': 'application/x-www-form-urlencoded',
+    },
+    body: new URLSearchParams({
+      client_id: "hasura-app",
+      username,
+      password,
+      grant_type: "password",
+      client_secret: "9ca6e96d-f72e-4208-91f4-a2d8e681f767",
+    }),
+  });
```
Comment (sagarkoshti1990) [inline src/services/auth/auth.ts:18]: Why are we using `fetch`? Please use `axios` instead.

File: src/utils/theme.tsx [hunk 4]
```diff
@@ -1,21 +1,108 @@
...
+      500: "#3182ce",
       900: "#1a202c",
+      primary: "#03627C",
+      fontColor: "#10162E",
+      greyColor: "#4F4F4F",
+      seaBlue: "#03627C1A",
+      lightGrey: "#F4F4F4",
+      lightBlue: "#39AED8"
+
+    },
```
Comment (sagarkoshti1990) [inline src/utils/theme.tsx:25]: The theme color name needs to be reset.

File: src/utils/theme.tsx [hunk 5]
```diff
@@ -1,21 +1,108 @@
...
+      baseStyle: {
+        minH: "100vh",
+        display: "flex",
+        alignItems: "center",
+        justifyContent: "center",
+      },
+    },
+    Box: {
+      baseStyle: {
+        maxW: "350px",
```
Comment (sagarkoshti1990) [inline src/utils/theme.tsx:38]: Why are we using a hardcoded max width of 350px?

File: src/utils/theme.tsx [hunk 6]
```diff
@@ -1,21 +1,108 @@
...
+        w: "100%",
+        py: 8,
+        px: 4,
+        borderRadius: "md",
+        boxShadow: "md",
+      },
+    },
+    ImageBox: {
+      baseStyle: {
+        width: "200px",
```
Comment (sagarkoshti1990) [inline src/utils/theme.tsx:48]: Why are we using a hardcoded max width of 200px?

**Review by sagarkoshti1990 — CHANGES_REQUESTED** [review]

**Comment by sagarkoshti1990** [comment]
The file `src/assets/logo/bg.png` should be in `src/assets/images/bg.png`.

File: src/services/auth/auth.ts (hunk 3 above)
Comment (deepali-chavhan) [reply src/services/auth/auth.ts:18]: I get a CORS error if I use Axios. I use fetch to avoid this.

File: src/components/common/PopupModal.tsx [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+          <Typography variant="h2" color={theme.colors.brand.textSecondary} fontSize="14px" style={{ paddingLeft: '28px' }}><strong>MM:</strong> {t("MONTH")} </Typography>
+          <Typography variant="h2" color={theme.colors.brand.textSecondary} fontSize="14px" style={{ paddingLeft: '28px' }}><strong>YYYY:</strong> {t("YEAR")} </Typography>
+        </Box>
+
+      </Box>
+
+      {/* Example */}
+      <h2 style={{ color: theme.colors.brand.textSecondary, fontSize: '12px', marginTop: '10px', marginBottom: '10px', backgroundColor: theme.colors.brand.backgroundHighlight, padding: '10px' }}>
+        <strong>{t("EXAMPLE")}:</strong><i>{example}</i> <strong>‘ANKU30121988’</strong>
+      </h2>
```
Comment (sagarkoshti1990) [inline src/components/common/PopupModal.tsx:119]: The Typography component is not a single component; it should include H1, H2, ..., H6, etc.

File: src/components/common/button/CommonButton.tsx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+  style?: React.CSSProperties;
+  children: React.ReactNode;
+  className?: string;
+  disabled?: boolean;
+  color?: string;
+  backgroundColor?: string;
+}
+
+
+const CommonButton: React.FC<CommonButtonProps> = ({
```
Comment (sagarkoshti1990) [inline src/components/common/button/CommonButton.tsx:14]: The common button component should be named PrimaryButton, SecondaryButton, etc.

File: src/components/common/typography/Typography.tsx [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+  variant?: 'h1' | 'h2' | 'h3' | 'p';
+  color?: string;
+  fontSize?: string;
+  fontWeight?: string;
+  marginBottom?: string;
+  className?: string;
+  style?: React.CSSProperties;
+}
+
+const Typography: React.FC<TypographyProps> = ({
```
Comment (sagarkoshti1990) [inline src/components/common/typography/Typography.tsx:15]: The Typography component is not a single component; it should include H1, H2, ..., H6, etc.

File: src/pages/Home.tsx [hunk 10]
```diff
@@ -1,15 +1,154 @@
...
+import background from '../assets/images/home-bg.png';
+import transformSchool from '../assets/logo/SG_Transform Schools_Logo.png';
+import { useTranslation } from 'react-i18next';
+import { getProgramId, getSubjectList } from "../services/home/home";
+import english from '../assets/icons/english_icon.png';
+import physics from '../assets/icons/physics_icon.png';
+import math from '../assets/icons/maths_icon.png';
+import kannada from '../assets/icons/kannada_icon.png';
+
+export default function Homepage() {
```
Comment (sagarkoshti1990) [inline src/pages/Home.tsx:28]: Please remove the homepage component from this PR and add it in a new PR.

File: src/pages/auth/Login.tsx [hunk 11]
```diff
@@ -1,67 +1,280 @@
...
+          //   localStorage.setItem("fcmToken", fcmToken);
+          // } catch (e) {
+          //   localStorage.setItem("fcmToken", "");
+          // }
+          // eventBus.publish("AUTH", {
+          //   eventType: "LOGIN_SUCCESS",
+          //   data: {
+          //     token: token,
+          //   },
+          // });
```
Comment (sagarkoshti1990) [inline src/pages/auth/Login.tsx:170]: Please remove the commented-out code.

File: src/services/home/home.ts [hunk 12]
```diff
@@ -0,0 +1,85 @@
+import moment from 'moment'
+import _ from 'lodash';
+const dateFor = moment().format('YYYY-MM-DD')
+
+export const getProgramId = async () => {
```
Comment (sagarkoshti1990) [inline src/services/home/home.ts:5]: Please remove the home service from this PR and add it in a new PR.

File: src/utils/theme.tsx [hunk 13]
```diff
@@ -1,21 +1,105 @@
...
+      primary: "#03627C",           // Primary brand color for prominent UI elements
+      primaryLight: "#39AED8",      // Lighter version of the primary color, for highlights
+
+      textPrimary: "#10162E",       // Primary font color for main text
+      textSecondary: "#4F4F4F",     // Secondary font color for less prominent text
+
+      divider: "#3182ce",           // Color for dividers or lines between elements
+      backgroundHighlight: "#03627C1A", // Semi-transparent blue for subtle highlights
+      backgroundGrey: "#F4F4F4",    // Light grey for background sections
+      white: "#FFFFFF",             // Pure white, useful for text or backgrounds
```
Comment (sagarkoshti1990) [inline src/utils/theme.tsx:26]: This is also not the correct way to use the color theme.

File: src/components/common/PopupModal.tsx [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+
+      <Box
+
+      >
+        <Box display="flex" justifyContent="center" mb={4}>
+          {['FF', 'LL', 'DD', 'MM', 'YYYY'].map((item) => (
+            <Box
+              key={item}
+              p="5px 10px"
+              bg={theme.colors.brand.backgroundGrey}
```
Comment (sagarkoshti1990) [inline src/components/common/PopupModal.tsx:97]: This is also not the correct way to use the color theme.

**Review by sagarkoshti1990 — CHANGES_REQUESTED** [review]

**Pull request merged**